diff --git a/src/http/v3/ngx_http_v3.c b/src/http/v3/ngx_http_v3.c
index 77f04f1..de3d044 100644
--- a/src/http/v3/ngx_http_v3.c
+++ b/src/http/v3/ngx_http_v3.c
@@ -462,9 +462,6 @@ ngx_http_v3_handler(ngx_connection_t *c)
             case QUICHE_H3_EVENT_PRIORITY_UPDATE:
                 break;
 
-            case QUICHE_H3_EVENT_DATAGRAM:
-                break;
-
             case QUICHE_H3_EVENT_GOAWAY:
                 break;
         }
