diff --git a/src/http/ngx_http_request.c b/src/http/ngx_http_request.c
index 1f98cda..c90704a 100644
--- a/src/http/ngx_http_request.c
+++ b/src/http/ngx_http_request.c
@@ -822,7 +822,7 @@ ngx_http_ssl_handshake_handler(ngx_connection_t *c)

         c->ssl->no_wait_shutdown = 1;

-#if (NGX_HTTP_V2                                                              \
+#if ((NGX_HTTP_V2 || NGX_HTTP_V3)                                             \
      && defined TLSEXT_TYPE_application_layer_protocol_negotiation)
         {
         unsigned int            len;
